lib/Basics/files.cpp

```diff
@@ -1973,20 +1962,24 @@ static std::string getTempPath() {
 }
 
 static int mkDTemp(char* s, size_t bufferSize) {
-  std::string out;
-  icu::UnicodeString sw(s);
-  auto w = std::make_unique<wchar_t[]>(bufferSize);
-  static_assert(sizeof(wchar_t) == sizeof(char16_t), "icu utf16 type needs to match wchar_t");
-  memcpy(w.get(), sw.getTerminatedBuffer(), sizeof(wchar_t) * bufferSize);
-  // this will overwrite the _XXX part of the string:
-  auto rc = _wmktemp_s(w.get(), bufferSize);
-  if (rc == 0) {
+  std::string tmp(s, bufferSize);
+  std::wstring ws = toWString(tmp);
+
+  // get writeable copy of wstring buffer and replace the _XXX part in the buffer
+  std::vector<wchar_t> writeBuffer;
+  writeBuffer.resize(ws.size());
+  memcpy(writeBuffer.data(), ws.data(), sizeof(wchar_t) * ws.size());
+  auto rc = _wmktemp_s(writeBuffer.data(), writeBuffer.size());  // requires writeable buffer -- returns errno_t
+
+  if (rc == 0) {  // error of 0 is ok
     // if it worked out, we need to return the utf8 version:
-    sw = w.get();
-    sw.toUTF8String<std::string>(out);
-    memcpy(s, out.c_str(), bufferSize);
+    ws = std::wstring(writeBuffer.data(), writeBuffer.size());  // write back to wstring
+    tmp = fromWString(ws);
+    memcpy(s, tmp.data(), bufferSize);  // copy back into parameter
```

Comment:
is the size of `tmp` guaranteed to be `>= bufferSize` here?

Reply:
it is exactly equal

Reply:
ok